### PR TITLE
Change/tree usability

### PR DESCRIPTION
**Pull request by ist-rw**

### CODE REVIEW

File: sling/core/commons/src/main/java/com/composum/sling/core/servlet/NodeServlet.java [hunk 1]
```diff
@@ -806,111 +837,74 @@ public void doIt(SlingHttpServletRequest request, SlingHttpServletResponse respo
...
-
-                        jsonWriter.beginObject();
-                        jsonWriter.name("message").value("no modification");
-                        jsonWriter.endObject();
+                    Node oldParentNode = node.getParent();
+                    Integer index = params.index;
+                    if (index == null && oldParentNode != null && params.path.equals(oldParentNode.getPath())) {
+                        // preserve index in case of a simple rename
+                        NodeIterator nodes = oldParentNode.getNodes();
+                        for (index = 0; nodes.hasNext() && !oldPath.equals(nodes.nextNode().getPath()); index++) ;
```
Comment (mzeibig) [inline sling/core/commons/src/main/java/com/composum/sling/core/servlet/NodeServlet.java]: for loop with empty body

File: sling/core/commons/src/main/java/com/composum/sling/core/util/RequestUtil.java [hunk 2]
```diff
@@ -131,6 +131,15 @@ public static int getParameter(SlingHttpServletRequest request, String name, int
         return result;
     }
 
+    public static boolean getParameter(SlingHttpServletRequest request, String name, boolean defaultValue) {
+        Boolean result = defaultValue;
+        String string = request.getParameter(name);
+        if (StringUtils.isNotBlank(string)) {
+            result = Boolean.parseBoolean(string);
+        }
+        return result != null ? result : defaultValue;
```
Comment (mzeibig) [inline sling/core/commons/src/main/java/com/composum/sling/core/util/RequestUtil.java]: always true

File: sling/core/commons/src/main/java/com/composum/sling/core/servlet/NodeServlet.java (hunk 1 above)
Comment (ist-rw) [inline sling/core/commons/src/main/java/com/composum/sling/core/servlet/NodeServlet.java]: yes, sure - i need the index and the API provides an iterator only

File: sling/core/commons/src/main/java/com/composum/sling/core/util/RequestUtil.java (hunk 2 above)
Comment (ist-rw) [inline sling/core/commons/src/main/java/com/composum/sling/core/util/RequestUtil.java]: result can be 'null' if the 'defaultValue' - this i a case used really in the code

**Pull request merged**